serializability-checker-compiler-plugin/src/main/scala/org/virtuslab/ash/SerializabilityCheckerCompilerPlugin.scala

```diff
@@ -10,17 +10,27 @@ class SerializabilityCheckerCompilerPlugin(override val global: Global) extends
     """checks whether a specified Akka serialization is applied to all messages, events and persistent state classes"""
 
   //Placeholder options
-  private val pluginOptions = new SerializabilityCheckerOptions(verbose = false)
+  private val pluginOptions = new SerializabilityCheckerOptions(
+    verbose = false,
+    detectionFromGenerics = true,
+    detectionFromGenericMethods = true,
+    detectionFromMethods = true)
 
   override val components: List[PluginComponent] = List(
     new SerializabilityCheckerCompilerPluginComponent(pluginOptions, global))
 
   override def init(options: List[String], error: String => Unit): Boolean = {
-    pluginOptions.verbose = options.contains("verbose")
+    pluginOptions.verbose = options.contains("--verbose")
+    pluginOptions.detectionFromGenerics = !options.contains("--disable-detection-generics")
+    pluginOptions.detectionFromGenericMethods = !options.contains("--disable-detection-generic-methods")
+    pluginOptions.detectionFromMethods = !options.contains("--disable-detection-methods")
     true
   }
 
   override val optionsHelp: Option[String] = Some("""
-      |verbose - print additional info about detected serializability traits and serializable classes
+      |--verbose - print additional info about detected serializability traits and serializable classes
+      |--disable-detection-generics - disables detection of classes using list of generic types
```

Comment:
🤔

Reply:
This semantics is pretty unclear here judging on solely the name/description TBH :/

Reply:
This gets much clearer ofc once I take a look into `SerializabilityCheckerCompilerPluginComponent`... but in principle we shouldn't let the users rely on the plugin's source code. Maybe provide some examples:

```suggestion
      |--disable-detection-generics - disables detection of messages/events/state based on their usage as a type param of certain classes, e.g. akka.actor.typed.Behavior or akka.persistence.typed.scaladsl.Effect
```

Reply:
Same for the two options below